File: 3_WebAppStudy/JAVA00_WebAppBoardModel1/src/main/java/com/example/webapptest/service/UsersService.java
package com.example.webapptest.service;


import com.example.webapptest.dto.UsersDto;

//기획 => 와이어프레임(ux ui) =>serivce => dao
//구현 => 반대로
public interface UsersService {
    UsersDto login(String uId, String pw); //로그인
    UsersDto detail(String uId); //상세정보
    UsersDto idCheck(String uId); //중복아이디 체크
    int modify(UsersDto user); //개인정보 수정
    int register(UsersDto user); //회원가입
    int closeAccount(UsersDto user); //비공개
    //회원탈퇴 =>삭제x, permission = "PRIVATE"
    int dropOut(String uId,String pw); //탈퇴(삭제)

}

File: 3_WebAppStudy/JS04_ExpressBoardStudy/model/service/UsersService.js
const pool=require("../db/WebAppBoardPool");
const UsersDao=require("../dao/UsersDao");
const usersDao=new UsersDao(pool);
const PageVo = require("../vo/PageVoTest");
class UsersService{
    async list(reqQuery){

            let users;
            let cnt = await usersDao.countBySearch(reqQuery.field, reqQuery.value);
            const pageVo  = new PageVo(cnt,reqQuery);
            users = await usersDao.findPage(pageVo);
            users.pageVo = pageVo;

        return users
    }
    async detail(uId){
        return usersDao.findById(uId);
    }

    async login(uId,pw){
        //로그인한 기록을 추가, pw 수정일을 확인
        return usersDao.findByUidAndPw(uId,pw);
    }


    async permissionModify(uId,permission){
        return usersDao.updatePermissionById(uId,permission)
    }
    async modify(user){
        return usersDao.updateById(user);
    }
    async register(user){
        return usersDao.insertOne(user);
    }
    async remove(uId){
        return usersDao.deleteOne(uId);
    }
}


module.exports=new UsersService(); //라우터에서 최종 사용!
